### ISSUE TITLE
Inconsistent behaviour

### ISSUE THREAD

**Issue by mehdi-ghezal** [PrestaShop/PrestaShop#11273]
**Describe the bug**

[In this line](https://github.com/PrestaShop/PrestaShop/blob/636770391b4c8cd4b9c115382bd7558007fbd1e7/classes/module/Module.php#L1134) we kill the application if we cannot validate the module name but only if `_PS_MODE_DEV_` is true. 

This PHP Constant is labeled as `Mode debug` in the back office but the name seems like a development mode. It's confusing... 

If `_PS_MODE_DEV_` is a development mode it should not be administrable by the back office and if the application is killed, we shouldn't be worry about a translate message but about show as many details as we can about the error.

If `_PS_MODE_DEV_` is a debug mode, the application shouldn't be killed and instead log the error.

Whatever the real use of `_PS_MODE_DEV_` the current behaviour of code I linked seems quite inconsistent.

**Additionnal information**
PrestaShop version: 1.7.x
PHP version: 7.1

**Comment by khouloudbelguith** [comment]
Hi @mehdi-ghezal,

Thanks for your report.
@eternoendless, @PrestaShop/prestashop-core-developers what do you think?
Thanks!

**Comment by MatShir** [comment]
Actually, _PS_MODE_DEV_ is used as debugging and development mode. You are right, we should have more error details. It will make sense to have exception variable from the class PrestaShopException instead.
But we can avoid the die function since we don’t really know what will happen if we remove it. It’s shared with our team @eternoendless. Also, don’t hesitate to make a pull request (https://github.com/PrestaShop/PrestaShop/tree/develop) if you have a code suggestion.

**Comment by eternoendless** [comment]
I agree on most of what you say @mehdi-ghezal, except for the option of toggling this mode in the back office, because it's useful for merchants when their shop breaks in production and they need to find out why. 

In any case, this is very old behavior and should be changed. I'm in favor of replacing the `die` with an exception so people can have a stack trace. And also, we shouldn't use the translator on debug messages.
